File: app/src/main/java/com/chiului/android_mvvm_architecture/bean/UserBean.java
package com.chiului.android_mvvm_architecture.bean;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

/**
 * 用户实体类$
 *
 * @author 神经大条蕾弟
 * @date 2020/07/21 09:56
 */
@Entity(tableName = "user")
public class UserBean {

    @NonNull
    @PrimaryKey
    private String token;
    private String id;
    private String avatar;
    private String mobile;
    private String username;// 用户名
    private String realName;// 真实姓名
    private String nickName;// 昵称
    private int sex;

    @NonNull
    public String getToken() {
        return token;
    }

    public void setToken(@NonNull String token) {
        this.token = token;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getAvatar() {
        return avatar;
    }

    public void setAvatar(String avatar) {
        this.avatar = avatar;
    }

    public String getMobile() {
        return mobile;
    }

    public void setMobile(String mobile) {
        this.mobile = mobile;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getRealName() {
        return realName;
    }

    public void setRealName(String realName) {
        this.realName = realName;
    }

    public String getNickName() {
        return nickName;
    }

    public void setNickName(String nickName) {
        this.nickName = nickName;
    }

    public int getSex() {
        return sex;
    }

    public void setSex(int sex) {
        this.sex = sex;
    }
}

File: app/src/main/java/com/chiului/android_mvvm_architecture/bean/UserBean.kt
package com.chiului.android_mvvm_architecture.bean

import androidx.annotation.NonNull
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * 用户实体类$
 * @author    神经大条蕾弟
 * @date      2021/02/01 15:16
 */
@Entity(tableName = "user")
data class UserBean(
        @NonNull
        @PrimaryKey
        var token: String,
        var id: String?,
        var avatar: String?,
        var mobile: String?,
        var username: String?,// 用户名
        var realName: String?,// 真实姓名
        var nickName: String?,// 昵称
        var sex: Int
)
